genesyscloud/task_management_workitem_schema/genesyscloud_task_management_workitem_schema_proxy.go

```diff
@@ -0,0 +1,226 @@
+func getTaskManagementProxy(clientConfig *platformclientv2.Configuration) *taskManagementProxy {
+	if internalProxy == nil {
+		internalProxy = newTaskManagementProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// createTaskManagementWorkitemSchema creates a Genesys Cloud task management workitem schema
+func (p *taskManagementProxy) createTaskManagementWorkitemSchema(ctx context.Context, schema *platformclientv2.Dataschema) (*platformclientv2.Dataschema, error) {
+	return p.createTaskManagementWorkitemSchemaAttr(ctx, p, schema)
+}
+
+// getAllTaskManagementWorkitemSchema retrieves all Genesys Cloud task management workitem schemas
+func (p *taskManagementProxy) getAllTaskManagementWorkitemSchema(ctx context.Context) (*[]platformclientv2.Dataschema, error) {
+	return p.getAllTaskManagementWorkitemSchemaAttr(ctx, p)
+}
+
+// getTaskManagementWorkitemSchemaIdByName returns a single Genesys Cloud task management workitem schema by a name
+func (p *taskManagementProxy) getTaskManagementWorkitemSchemasByName(ctx context.Context, name string) (schemas *[]platformclientv2.Dataschema, retryable bool, err error) {
+	return p.getTaskManagementWorkitemSchemasByNameAttr(ctx, p, name)
+}
+
+// getTaskManagementWorkitemSchemaById returns a single Genesys Cloud task management workitem schema by Id
+func (p *taskManagementProxy) getTaskManagementWorkitemSchemaById(ctx context.Context, id string) (schema *platformclientv2.Dataschema, statusCode int, err error) {
+	return p.getTaskManagementWorkitemSchemaByIdAttr(ctx, p, id)
+}
+
+// updateTaskManagementWorkitemSchema updates a Genesys Cloud task management workitem schema
+func (p *taskManagementProxy) updateTaskManagementWorkitemSchema(ctx context.Context, id string, schemaUpdate *platformclientv2.Dataschema) (*platformclientv2.Dataschema, error) {
+	return p.updateTaskManagementWorkitemSchemaAttr(ctx, p, id, schemaUpdate)
+}
+
+// deleteTaskManagementWorkitemSchema deletes a Genesys Cloud task management workitem schema by Id
+func (p *taskManagementProxy) deleteTaskManagementWorkitemSchema(ctx context.Context, id string) (statusCode int, err error) {
+	return p.deleteTaskManagementWorkitemSchemaAttr(ctx, p, id)
+}
+
+// getTaskManagementWorkitemSchemaDeletedStatus gets the deleted status of a Genesys Cloud task management workitem schema
+func (p *taskManagementProxy) getTaskManagementWorkitemSchemaDeletedStatus(ctx context.Context, schemaId string) (isDeleted bool, statusCode int, err error) {
+	return p.getTaskManagementWorkitemSchemaDeletedStatusAttr(ctx, p, schemaId)
+}
+
+// createTaskManagementWorkitemSchemaFn is an implementation function for creating a Genesys Cloud task management workitem schema
+func createTaskManagementWorkitemSchemaFn(ctx context.Context, p *taskManagementProxy, schema *platformclientv2.Dataschema) (*platformclientv2.Dataschema, error) {
+	createdSchema, _, err := p.taskManagementApi.PostTaskmanagementWorkitemsSchemas(*schema)
```

Comment:
Lets put a log message before the create and after the create that logs by name the schema being created.  Immediately after the call log by name, resp status code, error and correlation id that the call was created successfully.  The reason why I want to have this is that often times when I am working with the PS team where they are exporting mass numbers of items they might have a failure and they only know the failed item by name.  I want to log the correlation id to because if there is a failure they can quickly see the correlation id rather then digging through the SDK debug log and trying to find the specific all.

Reply:
Added logging for create and also noted this for future resources. I guess my question is do we want the detailed logging(with correlation id) on the other crud methods as well?

Reply:
Right now I am only putting it on the create because that's where the problems tend to be.  Declan is reviewing the new golang slog capability so we might be revamping a lot of our coding work anyways